### PR TITLE
fix: make operations with reference key safe

### PR DESCRIPTION
**Pull request by nikpivkin**
See https://github.com/aquasecurity/trivy/issues/4983

### CODE REVIEW

File: pkg/terraform/reference.go [hunk 1]
```diff
@@ -133,33 +132,43 @@ func (r Reference) RefersTo(a Reference) bool {
 }
 
 func (r *Reference) SetKey(key cty.Value) {
+	if key.IsNull() || !key.IsKnown() {
```
Comment (simar7) [inline pkg/terraform/reference.go:135]: Wouldn't this panic if key was actually null? In that case the `!key.IsKnown()` would panic.

Comment (simar7) [reply pkg/terraform/reference.go:135]: ping @nikpivkin

Comment (nikpivkin) [reply pkg/terraform/reference.go:135]: @simar7 The right operand of a boolean expression is evaluated [conditionally](https://go.dev/ref/spec#Logical_operators). If the `key` is null, then the left operand of the expression in the condition will be true and the function will exit, without checking whether the `key` is known.  And in any case, why should this check cause panic?

Comment (simar7) [reply pkg/terraform/reference.go:135]: I see. I didn’t know the conditional evaluation of Go for such a case. This seems to follow the short circuit behaviour. 

Thanks this works fine then.

**Review by simar7 — APPROVED** [review]

**Pull request merged**